main.py

```diff
@@ -1721,6 +1721,14 @@ def show_hover(self, point, contents):
             max_width=800)
 
 
+def format_popup_content(contents: str) -> str:
```

Comment:
To avoid confusion about what "formatting" really means, could you rename this to something like `preserve_whitespace` or `encode_whitespace` (or something else to your liking?)